core/sdk-core/src/main/java/software/amazon/awssdk/core/retry/RetryMode.java

```diff
@@ -84,9 +84,32 @@ public enum RetryMode {
      * the same client. When using adaptive retry mode, we recommend using a single client per resource.
      *
      * @see RetryPolicy#isFastFailRateLimiting()
+     * @deprecated As of 2.25.xx, replaced by {@link #ADAPTIVE_V2}. The ADAPTIVE implementation has a bug that prevents it
+     * from remembering its state across requests which is needed to correctly estimate its sending rate. Given that the
+     * this bug has been present since its introduction and that correct version might change the traffic patterns of the SDK we
+     * deemed too risky to fix this implementation..
      */
+    @Deprecated
     ADAPTIVE,
 
+    /**
+     * Adaptive V2 retry mode builds on {@link #STANDARD} mode.
+     * <p>
+     * Adaptive retry mode dynamically limits the rate of AWS requests to maximize success rate. This may be at the
+     * expense of request latency. Adaptive V2 retry mode is not recommended when predictable latency is important.
+     * <p>
+     * Adaptive V2 mode differs from {@link #ADAPTIVE} mode in the computed delays between calls, including the fist attempt
+     * that might be delayed if the algorithm considers that it's needed to increase the odds of a successful response. The
+     * previous Adaptive mode is also designed to work like so but a bug present since its introduction prevents it from
+     * preserving its internal state across requests.
+     * <p>
```

Comment:
s/fist/first/

Also should "Adaptive" be capitalized?

```
The
     * previous Adaptive mode is also designed to work like so but a bug present since its introduction prevents it from
     * preserving its internal state across requests.
```

I don't think it's necessary to specify that `ADAPTIVE` is also supposed to work this. If we link back to `ADAPTIVE` from here I think that's sufficient